lib/identity/client.js

```diff
@@ -793,16 +862,29 @@ class Client extends PartialClient {
         await isValidToznySecretNamespace(groupList.groups[index].groupName)
       ) {
         // check if they have any records shared with them
-        const recordsShared = await this.storage.listRecordsSharedWithGroup(
-          groupList.groups[index].groupID
-        )
+        let recordsShared
+        try {
+          recordsShared = await this.storage.listRecordsSharedWithGroup(
+            groupList.groups[index].groupID
+          )
+        } catch (e) {
+          continue
+        }
         if (recordsShared.length > 0) {
           // Check to make sure they are the writer
           // and secret type and secret name are the same
-          groupSharedList.push({
-            username: groupList.groups[index].description,
-            groupMembers: groupList.groups[index].memberCount,
-          }) // add their username to the list which is stored in the group description
+          if (
+            this.validGroupForSecretSharedList(
+              groupList.groups[index],
+              secretName,
+              secretType
+            )
+          ) {
+            groupSharedList.push({
+              username: groupList.groups[index].description,
```

Comment:
This only exists if its between a user to user not for a namespace since we add to the namespace we dont create a new group between the users